spring-kafka/src/main/java/org/springframework/kafka/listener/SeekToCurrentErrorHandler.java

```diff
@@ -82,16 +91,58 @@ public SeekToCurrentErrorHandler(BiConsumer<ConsumerRecord<?, ?>, Exception> rec
 	 * @since 2.2
 	 */
 	public SeekToCurrentErrorHandler(@Nullable BiConsumer<ConsumerRecord<?, ?>, Exception> recoverer, int maxFailures) {
-		this.failureTracker = new FailedRecordTracker(recoverer, maxFailures, logger);
+		this.failureTracker = new FailedRecordTracker(recoverer, maxFailures, LOGGER);
+	}
+
+	/**
+	 * Whether the offset for a recovered record should be committed.
+	 * @return true to commit recovered record offsets.
+	 * @since 2.2.4
+	 */
+	protected boolean isCommitRecovered() {
+		return this.commitRecovered;
+	}
+
+	/**
+	 * Set to true to commit the offset for a recovered record. The container
+	 * must be configured with {@link AckMode#MANUAL_IMMEDIATE}. Whether or not
+	 * the commit is sync or async depends on the container's syncCommits
+	 * property.
+	 * @param commitRecovered true to commit.
+	 * @since 2.2.4
+	 * @see #setOffsetCommitCallback(OffsetCommitCallback)
+	 */
+	public void setCommitRecovered(boolean commitRecovered) {
+		this.commitRecovered = commitRecovered;
 	}
 
 	@Override
 	public void handle(Exception thrownException, List<ConsumerRecord<?, ?>> records,
 			Consumer<?, ?> consumer, MessageListenerContainer container) {
 
-		if (!SeekUtils.doSeeks(records, consumer, thrownException, true, this.failureTracker::skip, logger)) {
+		if (!SeekUtils.doSeeks(records, consumer, thrownException, true, this.failureTracker::skip, LOGGER)) {
 			throw new KafkaException("Seek to current after exception", thrownException);
 		}
+		else if (this.commitRecovered) {
+			if (container.getContainerProperties().getAckMode().equals(AckMode.MANUAL_IMMEDIATE)) {
+				ConsumerRecord<?, ?> record = records.get(0);
+				if (container.getContainerProperties().isSyncCommits()) {
+					consumer.commitSync(Collections.singletonMap(new TopicPartition(record.topic(), record.partition()),
+							new OffsetAndMetadata(record.offset() + 1)));
+				}
+				else {
+					OffsetCommitCallback commitCallback = container.getContainerProperties().getCommitCallback();
+					if (commitCallback == null) {
+						commitCallback = LOGGING_COMMIT_CALLBACK;
+					}
+					consumer.commitAsync(Collections.singletonMap(new TopicPartition(record.topic(), record.partition()),
+							new OffsetAndMetadata(record.offset() + 1)), commitCallback);
```

Comment:
Looks like `TopicPartition` and `OffsetAndMetadata` could be extracted as variables before `if..else`. We are going to use them in both cases anyway.